cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -49,35 +49,45 @@ const BootstrapUser = "token-bootstrap-client"
 // RetrieveValidatedConfigInfo connects to the API Server and tries to fetch the cluster-info ConfigMap
 // It then makes sure it can trust the API Server by looking at the JWS-signed tokens and (if CACertHashes is not empty)
 // validating the cluster CA against a set of pinned public keys
-func RetrieveValidatedConfigInfo(cfg *kubeadmapi.Discovery, timeout time.Duration) (*clientcmdapi.Config, error) {
-	return retrieveValidatedConfigInfo(nil, cfg, constants.DiscoveryRetryInterval, timeout)
+func RetrieveValidatedConfigInfo(dryRunClient clientset.Interface, cfg *kubeadmapi.Discovery, timeout time.Duration) (*clientcmdapi.Config, error) {
+	isDryRun := dryRunClient != nil
+	isTesting := false
```

Comment:
What's the purpose of `isTesting`? Can we remove it?

Reply:
it's a left over from when i was testing with different ways to make the changes in token.go. i will remove it.

Reply:
actually, it has a purpose. without it the function retrieveValidatedConfigInfo is difficult to unit test.
during tests retrieveValidatedConfigInfo gets isDryRun = false and isTesting = true, that results in no loading of additional kubeconfig files (retrieveValidatedConfigInfo loads kubeconfig files). during tests client is a fake client and during real usage client is a dryRun client.

ideas to improve this are welcome, but i think it might be suitable for a new PR that would refactor / break apart  retrieveValidatedConfigInfo. perhaps in multiple functions.